pkg/controllers/updaterun/controller_test.go

```diff
@@ -312,3 +312,145 @@ func TestHandleClusterApprovalRequest(t *testing.T) {
+			},
+			shouldEnqueue: true,
+			queuedName:    "test-update-run",
+		},
+		"it should enqueue the targetUpdateRun, if ClusterApprovalRequest has only Approved status set to false": {
+			obj: &placementv1beta1.ClusterApprovalRequest{
+				ObjectMeta: metav1.ObjectMeta{
+					Generation: 1,
+				},
+				Spec: placementv1beta1.ApprovalRequestSpec{
+					TargetUpdateRun: "test-update-run",
+				},
+				Status: placementv1beta1.ApprovalRequestStatus{
+					Conditions: []metav1.Condition{
+						{
+							Status:             metav1.ConditionTrue,
+							Type:               string(placementv1beta1.ApprovalRequestConditionApproved),
+							ObservedGeneration: 1,
+						},
+					},
+				},
+			},
+			shouldEnqueue: true,
+			queuedName:    "test-update-run",
+		},
+		"it should not enqueue updateRun, if ClusterApprovalRequest has Approved set to false, ApprovalAccepted status set to true": {
+			obj: &placementv1beta1.ClusterApprovalRequest{
+				ObjectMeta: metav1.ObjectMeta{
+					Generation: 1,
+				},
+				Spec: placementv1beta1.ApprovalRequestSpec{
+					TargetUpdateRun: "test-update-run",
+				},
+				Status: placementv1beta1.ApprovalRequestStatus{
+					Conditions: []metav1.Condition{
+						{
+							Status:             metav1.ConditionFalse,
+							Type:               string(placementv1beta1.ApprovalRequestConditionApproved),
+							ObservedGeneration: 1,
+						},
+						{
+							Status:             metav1.ConditionTrue,
+							Type:               string(placementv1beta1.ApprovalRequestConditionApprovalAccepted),
+							ObservedGeneration: 1,
+						},
+					},
+				},
+			},
+			shouldEnqueue: false,
+		},
+		"it should not enqueue updateRun, if ClusterApprovalRequest has Approved, ApprovalAccepted status set to true": {
+			obj: &placementv1beta1.ClusterApprovalRequest{
+				ObjectMeta: metav1.ObjectMeta{
+					Generation: 1,
+				},
+				Spec: placementv1beta1.ApprovalRequestSpec{
+					TargetUpdateRun: "test-update-run",
+				},
+				Status: placementv1beta1.ApprovalRequestStatus{
+					Conditions: []metav1.Condition{
+						{
+							Status:             metav1.ConditionTrue,
+							Type:               string(placementv1beta1.ApprovalRequestConditionApproved),
+							ObservedGeneration: 1,
+						},
+						{
+							Status:             metav1.ConditionTrue,
+							Type:               string(placementv1beta1.ApprovalRequestConditionApprovalAccepted),
+							ObservedGeneration: 1,
+						},
+					},
+				},
+			},
+			shouldEnqueue: false,
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			queue := &controllertest.Queue{TypedInterface: workqueue.NewTypedRateLimitingQueue[reconcile.Request](
+				workqueue.DefaultTypedItemBasedRateLimiter[reconcile.Request]())}
+			handleClusterApprovalRequestDelete(tt.obj, queue)
+			if got := queue.Len() != 0; got != tt.shouldEnqueue {
+				t.Fatalf("handleClusterApprovalRequestDelete() shouldEnqueue test `%s` got %t, want %t", name, got, tt.shouldEnqueue)
```

Comment:
Do we need to put the name again in the error log? As t.Run() has the name input. Not sure how the error log looks like.